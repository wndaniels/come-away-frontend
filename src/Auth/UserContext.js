import React from "react";
/**
 *  Context:
 *
 * Provides currentUser object and setter for it throughout app.
 */
const UserContext = React.createContext();

export default UserContext;
