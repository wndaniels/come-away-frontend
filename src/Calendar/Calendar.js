import { useState, useRef, useEffect, useContext } from "react";
import UserContext from "../Auth/UserContext.js";
import useLocalStorage from "../hooks/useLocalStorage.js";
import jwt from "jsonwebtoken";
import { DayPilot, DayPilotCalendar } from "@daypilot/daypilot-lite-react";
import ComeAwayApi from "../Api/api";

const styles = {
  wrap: {
    display: "flex",
  },
  left: {
    marginRight: "10px",
  },
  main: {
    flexGrow: "1",
  },
};

export const TOKEN_STORAGE_ID = "comeaway-token";

const Calendar = () => {
  const { currentUser, setCurrentUser } = useContext(UserContext);
  const [token, setToken] = useLocalStorage(TOKEN_STORAGE_ID);
  const [formError, setFormError] = useState([]);
  const [infoLoaded, setInfoLoaded] = useState(false);
  const calendarRef = useRef(null);
  const [day, setDay] = useState();
  const [month, setMonth] = useState();
  const [year, setYear] = useState();
  const [dueDate, setDueDate] = useState();
  const [startDateFromDueDate, setStartDateFromDueDate] = useState();
  const [calViewData, setCalViewData] = useState();
  const [calAvailBegin, setCalAvailBegin] = useState();
  const [calAvailEnd, setCalAvailEnd] = useState();
  const [calUserId, setCalUserId] = useState();
  const [calByUserId, setCalByUserId] = useState();
  const [visitorData, setVisitorData] = useState();
  const [visitorDataForDisplay, setVisitorDataforDisplay] = useState([]);

  const [update, setUpdate] = useState([]);

  const [calData, setCalData] = useState({
    viewType: calViewData,
    businessBeginsHour: calAvailBegin,
    businessEndsHour: calAvailEnd,
  });

  useEffect(
    function loadUserInfo() {
      async function getCurrentUser() {
        try {
          let { username } = jwt.decode(token);
          ComeAwayApi.token = token;
          let currentUser = await ComeAwayApi.getCurrentUser(username);
          setCurrentUser(currentUser);
          setInfoLoaded(true);
        } catch (errors) {
          setFormError(errors);
          setCurrentUser(null);
        }
      }

      setInfoLoaded(false);

      if (token) {
        getCurrentUser();
      }
    },
    [token, setCurrentUser]
  );

  useEffect(() => {
    async function getDueDate() {
      const dueDate = await ComeAwayApi.getAllDueDates();

      dueDate.forEach((d) => {
        if (currentUser?.id === d.userId) {
          setDueDate(d);
        }
      });
    }

    getDueDate();
  }, [currentUser?.id]);

  useEffect(
    function getDueDateData() {
      async function getDayData() {
        const dayData = await ComeAwayApi.getDays();

        dayData.forEach((d) => {
          if (dueDate?.dayId === d.id) {
            setDay(d.id);
          }
        });
      }

      async function getMonthData() {
        const monthData = await ComeAwayApi.getMonths();

        monthData.forEach((m) => {
          if (dueDate?.monthId === m.id) {
            setMonth(m.id);
          }
        });
      }

      async function getYearData() {
        const yearData = await ComeAwayApi.getYears();

        yearData.forEach((y) => {
          if (dueDate?.yearId === y.id) {
            setYear(y.year);
          }
        });
      }

      getDayData([]);
      getMonthData([]);
      getYearData([]);
    },
    [dueDate?.dayId, dueDate?.monthId, dueDate?.yearId]
  );

  useEffect(() => {
    async function createDueDateString() {
      let date;
      try {
        date = new Date(year + "-" + month + "-" + day).toISOString();
        setStartDateFromDueDate(date);
      } catch (errors) {
        return;
      }
    }
    createDueDateString();
  }, [year, month, day]);

  useEffect(() => {
    async function getCalDataByUser() {
      const userCalData = await ComeAwayApi.getAllCals();

      userCalData.forEach((d) => {
        if (currentUser.id === d.userId) {
          try {
            setCalByUserId(d);
            setCalViewData(d.viewType);
            setCalAvailBegin(d.businessBeginsHour);
            setCalAvailEnd(d.businessEndsHour);
            setCalUserId(d.userId);
          } catch (error) {
            return;
          }
        }
      });
    }

    async function getCalendarData() {
      let userCalendar;
      try {
        userCalendar = {
          viewType: calViewData,
          businessBeginsHour: calAvailBegin,
          businessEndsHour: calAvailEnd,
        };
      } catch (errors) {
        return;
      }

      setCalData(userCalendar);
    }

    getCalDataByUser([]);
    getCalendarData([]);
  }, [
    currentUser.id,
    calAvailBegin,
    calAvailEnd,
    calViewData,
    year,
    month,
    day,
  ]);

  useEffect(() => {
    async function getVisitorDataForDisplay() {
      const visitors = await ComeAwayApi.getAllVisitors();

      if (Array.isArray(visitors) && calByUserId?.id) {
        const matchingEvents = [];
        visitors.forEach((visitor) => {
          if (visitor.calendarId === calByUserId.id) {
            try {
              const event = {
                id: visitor.id,
                start: new DayPilot.Date(visitor.startTime),
                end: new DayPilot.Date(visitor.endTime),
                text: visitor.fullName,
              };
              matchingEvents.push(event);
            } catch (errors) {
              return;
            }
          }
        });
        if (matchingEvents.length > 0) {
          setVisitorDataforDisplay(matchingEvents);
        }
      }
    }

    async function getVisitorData() {
      const visitors = await ComeAwayApi.getAllVisitors();
      if (calByUserId?.id === visitors?.calendarId)
        try {
          visitors.forEach((v) => {
            setVisitorData(v);
          });
        } catch (errors) {
          return;
        }
    }

    getVisitorDataForDisplay();
    getVisitorData();
  }, [calByUserId?.id]);

  async function handleDeleteVisitors(args) {
    let username = currentUser.username;
    let id = args.e.data.id; // Get id from the deleted event

    try {
      await ComeAwayApi.deleteVisitor(id, username);
    } catch (errors) {
      return;
    }
  }

  async function onEventClick() {
    const modalHtml = `<div>
        <h2>Name: ${visitorData.fullName}</h2>
        <h2>Note: ${visitorData.note}</h2>
      </div>`;
    DayPilot.Modal.alert(modalHtml);
  }

  async function getPreviousWeek() {
    const dp = calendarRef.current.control;
    dp.startDate = dp.startDate.addDays(-7);
    dp.update();
  }

  async function getNextWeek() {
    const dp = calendarRef.current.control;
    dp.startDate = dp.startDate.addDays(7);
    dp.update();
  }

  return (
    <div style={styles.wrap}>
      <div style={styles.main}>
        <div className="week-change m-4">
          <button
            className="btn btn-sm btn-primary me-5"
            onClick={getPreviousWeek}
          >
            Previous
          </button>
          WEEK
          <button className="btn btn-sm btn-primary ms-5" onClick={getNextWeek}>
            Next
          </button>
        </div>
        <DayPilotCalendar
          viewType={"Week"}
          businessBeginsHour={calData.businessBeginsHour}
          businessEndsHour={calData.businessEndsHour}
          durationBarVisible={false}
          headerDateFormat={"dddd MM/dd/yy"}
          headerHeight={50}
          headerTextWrappingEnabled={true}
          allDayEvents={true}
          heightSpec={"BusinessHoursNoScroll"}
          hourWidth={80}
          cellHeight={40}
          startDate={startDateFromDueDate}
          startDay={startDateFromDueDate}
          events={visitorDataForDisplay}
          eventDeleteHandling={"Update"}
          onEventDeleted={handleDeleteVisitors}
          onEventClick={onEventClick}
          eventMoveHandling={"Disabled"}
          eventResizeHandling={"Disabled"}
          timeRangeSelectedHandling={"Disabled"}
          visibleStart={true}
          ref={calendarRef}
        />
      </div>
    </div>
  );
};

export default Calendar;
