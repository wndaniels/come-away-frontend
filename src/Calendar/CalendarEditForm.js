import React, { useContext, useEffect, useState } from "react";
import useLocalStorage from "../hooks/useLocalStorage.js";
import { useNavigate } from "react-router-dom";
import ComeAwayApi from "../Api/api.js";
import UserContext from "../Auth/UserContext.js";
import jwt from "jsonwebtoken";
import Alert from "../Common/Alert.js";
import DueDateEditForm from "../DueDate/DueDateEditForm.js";

export const TOKEN_STORAGE_ID = "comeaway-token";

const CalendarEditForm = () => {
  const navigate = useNavigate();
  const { currentUser, setCurrentUser } = useContext(UserContext);
  const [infoLoaded, setInfoLoaded] = useState(false);
  const [token, setToken] = useLocalStorage(TOKEN_STORAGE_ID);
  const [calAvailBegin, setCalAvailBegin] = useState([]);
  const [calAvailEnd, setCalAvailEnd] = useState([]);
  const [calByUser, setCalByUser] = useState([]);
  const [dueDateByUser, setDueDateByUser] = useState();
  const [formData, setFormData] = useState({
    businessBeginsHour: calByUser.businessBeginsHour,
    businessEndsHour: calByUser.businessEndsHour,
  });

  const [formError, setFormError] = useState([]);

  useEffect(
    function loadUserInfo() {
      async function getCurrentUser() {
        if (token) {
          try {
            let { username } = jwt.decode(token);
            ComeAwayApi.token = token;
            let currentUser = await ComeAwayApi.getCurrentUser(username);
            setCurrentUser(currentUser);
          } catch (errors) {
            setFormError(errors);
            setCurrentUser(null);
          }
        }
        setInfoLoaded(true);
      }
      setInfoLoaded(false);
      getCurrentUser();
    },
    [token, setCurrentUser]
  );

  useEffect(() => {
    async function getCalDataByUser() {
      const userCalData = await ComeAwayApi.getAllCals();
      try {
        userCalData.forEach((d) => {
          if (currentUser?.id === d.userId) setCalByUser(d);
        });
      } catch (errors) {
        return;
      }
    }
    getCalDataByUser([]);
  }, [currentUser?.id]);

  useEffect(() => {
    async function getDueDateByUser() {
      const dueDateData = await ComeAwayApi.getAllDueDates();

      try {
        dueDateData.forEach((d) => {
          if (currentUser?.id === d.userId) setDueDateByUser(d);
        });
      } catch (errors) {
        return;
      }
    }
    getDueDateByUser();
  }, [currentUser?.id]);

  useEffect(function getCalData() {
    async function getBeginHoursData() {
      try {
        const beginHourRes = await ComeAwayApi.getBeginHours();
        setCalAvailBegin(beginHourRes);
      } catch (errors) {
        return;
      }
    }

    async function getEndHourData() {
      try {
        const endHourRes = await ComeAwayApi.getEndHours();
        setCalAvailEnd(endHourRes);
      } catch (errors) {
        return;
      }
    }

    getBeginHoursData([]);
    getEndHourData([]);
  }, []);

  async function handleCalSubmit(evt) {
    evt.preventDefault();

    let calendarData = {
      businessBeginsHour: formData.businessBeginsHour,
      businessEndsHour: formData.businessEndsHour,
    };

    let username = currentUser.username;
    let id = calByUser.id;

    let updatedCal;

    try {
      updatedCal = await ComeAwayApi.updateCal(username, id, calendarData);
    } catch (errors) {
      setFormError([errors]);
      return;
    }
    navigate("/calendar");
    setFormData((f) => ({ ...f }));
    setFormError([]);
  }

  function handleIntChange(evt) {
    const { name, value } = evt.target;

    setFormData((f) => ({
      ...f,
      [name]: parseInt(value),
    }));

    setFormError([]);
  }

  async function handleDeleteCal(evt) {
    let username = currentUser.username;
    let id = calByUser.id;

    try {
      await ComeAwayApi.deleteCal(username, id);
    } catch (errors) {
      return;
    }
    navigate("/");
  }

  async function handleDeleteDueDate(evt) {
    let username = currentUser.username;
    let id = dueDateByUser.id;

    try {
      await ComeAwayApi.deleteDueDate(username, id);
    } catch (errors) {
      return;
    }
    navigate("/");
  }

  if (!infoLoaded) return;

  return (
    <div className="m-5">
      <div>
        <DueDateEditForm />
      </div>
      <div className="CalendarEditForm mt-5">
        <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
          <h1 className="mb-3">Edit Calendar</h1>
          <div className="card">
            <div className="card-body">
              <form method="patch">
                <div className="d-grid gap-3">
                  {formError.length ? (
                    <Alert
                      type="danger"
                      messages={["All fields must be complete."]}
                    ></Alert>
                  ) : null}

                  <div className="form-group">
                    <label htmlFor="businessBeginsHour">
                      * Select Start Time:
                    </label>
                    <select
                      name="businessBeginsHour"
                      className="form-control"
                      onChange={handleIntChange}
                    >
                      <option hidden>-</option>
                      {calAvailBegin &&
                        calAvailBegin.map((s, id) => (
                          <option value={s.businessBeginsHour} key={id}>
                            {s.hourTitle}
                          </option>
                        ))}
                    </select>
                  </div>

                  <div className="form-group">
                    <label htmlFor="businessEndsHour">* Select End Time:</label>
                    <select
                      name="businessEndsHour"
                      className="form-control mb-3"
                      onChange={handleIntChange}
                    >
                      <option hidden>-</option>
                      {calAvailEnd &&
                        calAvailEnd.map((e, id) => (
                          <option
                            defaultValue={e}
                            value={e.businessEndsHour}
                            key={id}
                          >
                            {e.hourTitle}
                          </option>
                        ))}
                    </select>
                  </div>
                </div>

                <button
                  onClick={handleCalSubmit}
                  className="btn btn-sm btn-primary "
                >
                  Update Calendar
                </button>
                <button
                  onClick={() => {
                    handleDeleteCal();
                    handleDeleteDueDate();
                  }}
                  className="btn btn-sm btn-danger ms-3"
                >
                  Delete
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CalendarEditForm;
