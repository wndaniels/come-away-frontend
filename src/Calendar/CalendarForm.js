import React, { useContext, useEffect, useState } from "react";
import useLocalStorage from "../hooks/useLocalStorage.js";
import { useNavigate } from "react-router-dom";
import ComeAwayApi from "../Api/api.js";
import UserContext from "../Auth/UserContext.js";
import jwt from "jsonwebtoken";
import Alert from "../Common/Alert.js";

export const TOKEN_STORAGE_ID = "comeaway-token";

const CalendarForm = () => {
  const navigate = useNavigate();
  const { currentUser, setCurrentUser } = useContext(UserContext);
  const [infoLoaded, setInfoLoaded] = useState(false);
  const [token, setToken] = useLocalStorage(TOKEN_STORAGE_ID);
  const [calAvailBegin, setCalAvailBegin] = useState([]);
  const [calAvailEnd, setCalAvailEnd] = useState([]);
  const [formData, setFormData] = useState({
    businessBeginsHour: 0,
    businessEndsHour: 0,
    userId: currentUser.id,
  });

  const [formError, setFormError] = useState([]);

  useEffect(
    function loadUserInfo() {
      async function getCurrentUser() {
        if (token) {
          try {
            let { username } = jwt.decode(token);
            ComeAwayApi.token = token;
            let currentUser = await ComeAwayApi.getCurrentUser(username);
            setCurrentUser(currentUser);
          } catch (errors) {
            setFormError(errors);
            setCurrentUser(null);
          }
        }
        setInfoLoaded(true);
      }
      setInfoLoaded(false);
      getCurrentUser();
    },
    [token, setCurrentUser]
  );

  useEffect(function getCalData() {
    async function getBeginHoursData() {
      const beginHourRes = await ComeAwayApi.getBeginHours();
      setCalAvailBegin(beginHourRes);
    }

    async function getEndHourData() {
      const endHourRes = await ComeAwayApi.getEndHours();
      setCalAvailEnd(endHourRes);
    }

    getBeginHoursData([]);
    getEndHourData([]);
  }, []);

  async function handleSubmit(evt) {
    evt.preventDefault();

    let calendarData = {
      businessBeginsHour: formData.businessBeginsHour,
      businessEndsHour: formData.businessEndsHour,
      userId: currentUser.id,
    };

    let username = currentUser.username;

    let createdCal;

    try {
      createdCal = await ComeAwayApi.createCal(username, calendarData);
    } catch (errors) {
      setFormError([errors]);
      return;
    }

    setFormData((f) => ({ ...f }));
    setFormError([]);
    navigate("/");
  }

  function handleChange(evt) {
    const { name, value } = evt.target;

    setFormData((f) => ({
      ...f,
      [name]: parseInt(value),
    }));

    setFormError([]);
  }

  if (!infoLoaded) return;

  return (
    <div className="CalendarForm mt-5">
      <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
        <h1 className="mb-3">Create Calendar</h1>
        <div className="card">
          <div className="card-body">
            <form method="post">
              <div className="d-grid gap-3">
                {formError.length ? (
                  <Alert
                    type="danger"
                    messages={["All fields must be complete."]}
                  ></Alert>
                ) : null}

                <div className="form-group">
                  <label htmlFor="businessBeginsHour">
                    * Select Start Time:
                  </label>
                  <select
                    name="businessBeginsHour"
                    className="form-control"
                    onChange={handleChange}
                  >
                    <option hidden>-</option>
                    {calAvailBegin &&
                      calAvailBegin.map((s, id) => (
                        <option value={s.businessBeginsHour} key={id}>
                          {s.hourTitle}
                        </option>
                      ))}
                  </select>
                </div>

                <div className="form-group">
                  <label htmlFor="businessEndsHour">* Select End Time:</label>
                  <select
                    name="businessEndsHour"
                    className="form-control mb-3"
                    onChange={handleChange}
                  >
                    <option hidden>-</option>
                    {calAvailEnd &&
                      calAvailEnd.map((e, id) => (
                        <option value={e.businessEndsHour} key={id}>
                          {e.hourTitle}
                        </option>
                      ))}
                  </select>
                </div>
              </div>

              <button onClick={handleSubmit} className="btn btn-sm btn-primary">
                Create Calendar
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CalendarForm;
