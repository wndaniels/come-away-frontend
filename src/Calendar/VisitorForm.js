import React, { useContext, useEffect, useState } from "react";
import useLocalStorage from "../hooks/useLocalStorage";
import { Form, useNavigate, useParams } from "react-router-dom";
import ComeAwayApi from "../api/api";
import UserContext from "../Auth/UserContext";
import jwt from "jsonwebtoken";
import Alert from "../Common/Alert";

// export const TOKEN_STORAGE_ID = "comeaway-token";

const VisitorForm = () => {
  const navigate = useNavigate();
  // const { currentUser, setCurrentUser } = useContext(UserContext);
  const [infoLoaded, setInfoLoaded] = useState(false);
  // const [token, setToken] = useLocalStorage(TOKEN_STORAGE_ID);
  const [paramUserId, setParamUserId] = useState([]);
  const [calId, setCalId] = useState([]);
  const [calAvailBegin, setCalAvailBegin] = useState([]);
  const [calAvailEnd, setCalAvailEnd] = useState([]);
  const [formData, setFormData] = useState({
    fullName: "",
    note: "",
    startTime: 0,
    endTime: 0,
    calendarId: calId.id,
  });

  const [visit, setVisit] = useState();
  const [formError, setFormError] = useState([]);

  const params = useParams();

  useEffect(
    function loadUsersInfo() {
      async function getUsers() {
        const userData = await ComeAwayApi.getAllUsers();
        try {
          userData.map((d) => {
            if (params.username === d.username) setParamUserId(d);
          });
        } catch (errors) {
          return;
        }
        setInfoLoaded(true);
      }
      setInfoLoaded(false);
      getUsers([]);
    },
    [params.username]
  );

  useEffect(() => {
    async function getCalIdByUser() {
      const calData = await ComeAwayApi.getCalData();
      try {
        calData.map((i) => {
          if (paramUserId.id === i.userId) setCalId(i);
        });
      } catch (errors) {
        return;
      }
    }
    getCalIdByUser([]);
  }, [paramUserId]);

  useEffect(function getCalData() {
    async function getBeginHoursData() {
      const beginHourRes = await ComeAwayApi.getBeginHours();
      setCalAvailBegin(beginHourRes);
    }

    async function getEndHourData() {
      const endHourRes = await ComeAwayApi.getEndtHours();
      setCalAvailEnd(endHourRes);
    }

    getBeginHoursData([]);
    getEndHourData([]);
  }, []);

  async function handleSubmit(evt) {
    evt.preventDefault();

    let visitData = {
      fullName: formData.fullName,
      note: formData.note,
      startTime: formData.startTime,
      endTime: formData.endTime,
      calendarId: calId.id,
    };

    let createdVisit;

    try {
      createdVisit = await ComeAwayApi.addVisit(visitData);
      console.log(createdVisit);
    } catch (errors) {
      setFormError([errors]);
      return;
    }
    // navigate("/calendar");
    setFormData((f) => ({ ...f }));
    setFormError([]);
    setVisit(createdVisit);
  }

  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData((f) => ({
      ...f,
      [name]: value,
    }));

    setFormError([]);
  }

  function handleHourChange(evt) {
    const { name, value } = evt.target;

    setFormData((f) => ({
      ...f,
      [name]: parseInt(value),
    }));

    setFormError([]);
  }

  if (!infoLoaded) return <h3>Loading...</h3>;

  return (
    <div className="CalendarForm">
      <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
        <h1 className="mb-3">Schedule a Visit!</h1>
        <div className="card">
          <div className="card-body">
            <Form method="patch" onSubmit={handleSubmit}>
              <div className="d-grid gap-3">
                <div className="form-group">
                  {formError.length ? (
                    <Alert
                      type="danger"
                      messages={["All fields must be complete."]}
                    ></Alert>
                  ) : null}
                  <div className="form-group">
                    <label htmlFor="fullName">* Name of Visitor(s):</label>
                    <input
                      name="fullName"
                      type="text"
                      onChange={handleChange}
                      className="form-control mb-3"
                    />
                  </div>
                  <div className="form-group">
                    <label htmlFor="note">Leave a note for the family:</label>
                    <textarea
                      name="note"
                      type="text"
                      onChange={handleChange}
                      className="form-control mb-3"
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="startTime">
                      * What time would you like to visit?:
                    </label>
                    <select
                      name="startTime"
                      className="form-control mb-3"
                      onChange={handleHourChange}
                    >
                      {calAvailBegin &&
                        calAvailBegin.map((s, id) => (
                          <option value={s.businessBeginsHour} key={id}>
                            {s.hourTitle}
                          </option>
                        ))}
                    </select>
                  </div>

                  <div className="form-group">
                    <label htmlFor="endTime">
                      * Select an end time for your visit.
                      <i> (Please try to limit visits to 1 hour)</i>:
                    </label>
                    <select
                      name="endTime"
                      className="form-control mb-3"
                      onChange={handleHourChange}
                    >
                      {calAvailEnd &&
                        calAvailEnd.map((e, id) => (
                          <option value={e.businessEndsHour} key={id}>
                            {e.hourTitle}
                          </option>
                        ))}
                    </select>
                  </div>
                </div>
              </div>

              <button className="btn btn-sm btn-primary">
                Update Calendar
              </button>
            </Form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default VisitorForm;
