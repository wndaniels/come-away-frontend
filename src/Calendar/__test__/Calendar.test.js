import "../../matchMedia.mock";
import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/jest-dom";
import Calendar from "../Calendar";
import UserContext from "../../Auth/UserContext";

jest.mock("@daypilot/daypilot-lite-react", () => ({
  ...jest.requireActual("@daypilot/daypilot-lite-react"),
  DayPilot: () => jest.fn(),
  DayPilotCalendar: () => jest.fn(),
}));

describe("Calendar", () => {
  it("renders without crashing", () => {
    const { asFragment } = render(
      <Wrapper>
        <Calendar />
      </Wrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("can find button text", () => {
    const { getByText } = render(
      <Wrapper>
        <Calendar />
      </Wrapper>
    );
    const previousButtonText = getByText(/previous/i);
    const nextButtonText = getByText(/next/i);
    expect(previousButtonText).toBeInTheDocument();
    expect(nextButtonText).toBeInTheDocument();
  });
});

function Wrapper({ children }) {
  return (
    <UserContext.Provider
      value={{
        currentUser: jest.fn(),
        setCurrentUser: jest.fn(),
        login: jest.fn(),
        signup: jest.fn(),
        logout: jest.fn(),
      }}
    >
      {children}
    </UserContext.Provider>
  );
}
