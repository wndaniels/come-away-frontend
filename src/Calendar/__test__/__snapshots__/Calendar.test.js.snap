// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar renders without crashing 1`] = `
<DocumentFragment>
  <div
    style="display: flex;"
  >
    <div
      style="flex-grow: 1;"
    >
      <div
        class="week-change m-4"
      >
        <button
          class="btn btn-sm btn-primary me-5"
        >
          Previous
        </button>
        WEEK
        <button
          class="btn btn-sm btn-primary ms-5"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
